api/v1beta1/kustomization_types.go

```diff
@@ -254,13 +256,35 @@ func SetKustomizationReadiness(k *Kustomization, status metav1.ConditionStatus,
 
 // KustomizationNotReady registers a failed apply attempt of the given Kustomization.
 func KustomizationNotReady(k Kustomization, revision, reason, message string) Kustomization {
+	RemoveKustomizationReconciling(&k)
+	SetKustomizationStalled(&k, metav1.ConditionTrue, reason, message)
 	SetKustomizationReadiness(&k, metav1.ConditionFalse, reason, trimString(message, MaxConditionMessageLength), revision)
 	if revision != "" {
 		k.Status.LastAttemptedRevision = revision
 	}
 	return k
 }
 
+// SetKustomizeStalled sets the StalledCondition, ObservedGeneration on the Kustomization.
+func SetKustomizationStalled(k *Kustomization, status metav1.ConditionStatus, reason, message string) {
+	meta.SetResourceCondition(k, meta.StalledCondition, status, reason, trimString(message, MaxConditionMessageLength))
+	k.Status.ObservedGeneration = k.Generation
+}
+
+// RemoveKustomizationStalled removes the StalledCondition on the Kustomization.
+func RemoveKustomizationStalled(k *Kustomization) {
+	if len(k.Status.Conditions) != 0 {
+		apimeta.RemoveStatusCondition(&k.Status.Conditions, meta.StalledCondition)
+	}
+}
+
+// RemoveKustomizationReconciling removes the ReconcilingCondition on the Kustomization.
+func RemoveKustomizationReconciling(k *Kustomization) {
+	if len(k.Status.Conditions) != 0 {
+		apimeta.RemoveStatusCondition(&k.Status.Conditions, meta.ReconcilingCondition)
+	}
+}
+
 // KustomizationNotReady registers a failed apply attempt of the given Kustomization,
 // including a Snapshot.
 func KustomizationNotReadySnapshot(k Kustomization, snapshot *Snapshot, revision, reason, message string) Kustomization {
```

Comment:
Shouldn't this also include the logic to remove and/or add the `Reconciling` and `Stalled` conditions?

Reply:
Yes it should cc @SomtochiAma